ddtrace/settings/_otel_remapper.py

```diff
@@ -0,0 +1,187 @@
+        )
+        return ""
+    return ""
+
+
+def _remap_otel_tags(otel_value):
+    """Remaps the otel tags to ddtrace tags"""
+    dd_tags = []
+    remaining_tags = []
+    otel_tags = otel_value.split(",")
+    otel_user_tag_dict = dict()
+
+    try:
+        for tag in otel_tags:
+            tag_pair = tag.split("=")
+            otel_user_tag_dict[tag_pair[0]] = tag_pair[1]
+
+        for otel_key, dd_key in OTEL_UNIFIED_TAG_MAPPINGS.items():
+            if otel_key in otel_user_tag_dict.keys():
+                dd_tags.append("{}:{}".format(dd_key, otel_user_tag_dict[otel_key]))
+
+        for key, value in otel_user_tag_dict.items():
+            if key.lower() not in OTEL_UNIFIED_TAG_MAPPINGS.keys():
+                if len(dd_tags) < 10:
+                    dd_tags.append("{}:{}".format(key, value))
+                else:
+                    remaining_tags.append("{}:{}".format(key, value))
+    except Exception:
+        log.warning("DDTRACE failed to read OTEL_RESOURCE_ATTRIBUTES. This value is misformatted: %s", otel_value)
+
+    if len(otel_user_tag_dict.items()) > 10:
+        log.warning(
+            "To preserve metrics cardinality, only the following first 10 tags have been processed %s. "
+            "The following tags were not ingested: %s",
+            dd_tags,
+            remaining_tags,
+        )
+    return ",".join(dd_tags)
+
+
+def _remap_otel_sdk_config(otel_value):
+    """Remaps the otel sdk config to ddtrace sdk config"""
+    if otel_value == "false":
+        return "True"
+    elif otel_value == "true":
+        return "False"
+    else:
+        log.warning("OTEL_SDK_DISABLED='%s'  is not supported", otel_value)
+        return otel_value
+
+
+def _remap_default(otel_value):
+    """Remaps the otel default value to ddtrace default value"""
+    return otel_value
+
+
+ENV_VAR_MAPPINGS = {
+    "OTEL_SERVICE_NAME": ("DD_SERVICE", _remap_default),
+    "OTEL_LOG_LEVEL": ("DD_TRACE_DEBUG", _remap_otel_log_level),
+    "OTEL_PROPAGATORS": ("DD_TRACE_PROPAGATION_STYLE", _remap_otel_propagators),
+    "OTEL_TRACES_SAMPLER": ("DD_TRACE_SAMPLE_RATE", _remap_traces_sampler),
+    "OTEL_TRACES_EXPORTER": ("DD_TRACE_ENABLED", _remap_traces_exporter),
+    "OTEL_METRICS_EXPORTER": ("DD_RUNTIME_METRICS_ENABLED", _remap_metrics_exporter),
+    "OTEL_LOGS_EXPORTER": ("", _remap_logs_exporter),  # Does not set a DDTRACE environment variable.
+    "OTEL_RESOURCE_ATTRIBUTES": ("DD_TAGS", _remap_otel_tags),
+    "OTEL_SDK_DISABLED": ("DD_TRACE_OTEL_ENABLED", _remap_otel_sdk_config),
+}
+
+
+def otel_remapping():
+    """Checks for the existence of both OTEL and Datadog tracer environment variables and remaps them accordingly.
+    Datadog Environment variables take precedence over OTEL, but if there isn't a Datadog value present,
+    then OTEL values take their place.
+    """
+    user_envs = {key.upper(): value for key, value in os.environ.items()}
```

Comment:
should they all be uppercase provided by the user anyways?

Reply:
@brettlangdon Hey Brett! We included that just to be safe in case a user supplied a mis-cased env var spelling-- would it be better to remove?

Reply:
no major issue, as long as there is a good reason to do so then 👍🏻